platforms/core-configuration/model-core/src/main/java/org/gradle/api/internal/provider/AbstractCollectionProperty.java

```diff
@@ -574,55 +644,40 @@ private ExecutionTimeValue<C> getFixedExecutionTimeValue(List<ExecutionTimeValue
 
         @Override
         public ValueProducer getProducer() {
-            return value.getProducer();
-        }
-
-        @Override
-        public String toString() {
-            return value.toString();
-        }
-    }
-
-    /**
-     * A provider for a collection type whose elements are themselves providers.
-     */
-    private static class CollectingProvider<T, C extends Collection<? extends T>> extends AbstractMinimalProvider<C> {
-        private final Class<C> type;
-        private final List<ProviderInternal<? extends Iterable<? extends T>>> providers;
-        private final Supplier<ImmutableCollection.Builder<T>> collectionFactory;
+            List<ValueProducer> producers = getCollectors().stream().map(ValueSupplier::getProducer).collect(toList());
+            return new ValueProducer() {
+                @Override
+                public void visitProducerTasks(Action<? super Task> visitor) {
+                    producers.forEach(c -> c.visitProducerTasks(visitor));
+                }
 
-        public CollectingProvider(Class<C> type, List<ProviderInternal<? extends Iterable<? extends T>>> providers, Supplier<ImmutableCollection.Builder<T>> collectionFactory) {
-            this.type = type;
-            this.providers = providers;
-            this.collectionFactory = collectionFactory;
-        }
+                @Override
+                public boolean isKnown() {
+                    return producers.stream().allMatch(ValueProducer::isKnown);
```

Comment:
🤔 `PlusProducer` was doing `left.isKnown() || right.isKnown();` Here we have `&&` effectively. Is it intended?

Reply:
Although no tests have caught that